### DISCUSSION TITLE
StylecopAnalyzers integration

### DISCUSSION THREAD

**Discussion by llob** [super-linter/super-linter#3516] [Ideas]
Hi,

We currently use super-linter extensively for linting everything from orchestration configuration files to C# and python. 
For C# though, dotnet-format really is not capable of catching the types of lint that I need it to (e.g., missing documentation, symbol naming, etc.). [StyleCopAnalyzers](https://github.com/DotNetAnalyzers/StyleCopAnalyzers) however seems to be able to do what I want, and having it supported in super-linter would therefore be very nice indeed. 

Assuming I do the work, is it feasible to get that kind of support merged into super-linter?

**Comment by nvuillam** [comment]
How would you install it on an alpine-based docker image ? :)

**Reply by llob** [reply]
I certainly have not looked into it in detail, but I assumed that since the project is based on .NET core, an approach similar to this [StyleCopAnalyzers.CLI](https://github.com/rookxx/StyleCopAnalyzers.CLI) would be plausible.
According to list of [supported distributions](https://learn.microsoft.com/en-us/dotnet/core/install/linux-alpine) from Microsoft, it should be doable.

I do not yet know much about the internals of either project, so if you have insights to share that would prevent this from working, I would love to learn?

**Comment by ferrarimarco** [comment]
Please, open a feature request for this, so we can track this effort. Also, PRs are more than welcome! Thanks :)
